modules/syslogformat/syslog-format.c

```diff
@@ -833,7 +833,8 @@ log_msg_parse_legacy(const MsgFormatOptions *parse_options,
       goto error;
     }
 
-  log_msg_parse_legacy_header(self, &src, &left, parse_options);
+  if ((parse_options->flags & LP_NO_HEADER) == 0)
```

Comment:
I think it's worth mentioning that `no-header` works only in `RFC3164`-mode, which means
- when `network(), tcp(), udp()` are used
- when `syslog()` or `flags(syslog-protocol)` is used (`RFC5424`), but the incoming message does not contain correct `pri` and `version` fields, so we fall back to `RFC3164`.

In other words, `flags(no-header)` does not work with properly formatted `RFC5424` messages.
@bazsi Should we address that? I can not really think of a good use-case, but the name `no-header` suggests that it should work in both cases.